instant-xml-macros/src/se.rs

```diff
@@ -67,42 +70,79 @@ impl<'a> Serializer {
                 output.extend(quote!(
-                    field.attribute = Some(instant_xml::FieldAttribute::Namespace(#namespace_key));
-                ));
+                        // Check if such namespace already exist, if so change it to use its prefix
+                        match serializer.parent_namespaces.get(#namespace) {
+                            Some(key) => field.attribute = Some(instant_xml::FieldAttribute::Prefix(key)),
+                            None => field.attribute = Some(instant_xml::FieldAttribute::Namespace(#namespace)),
+                        };
+                    )
+                );
             }
             Some(FieldAttribute::PrefixIdentifier(prefix_key)) => {
+                match self.other_namespaces.get(&prefix_key) {
+                    Some(val) => {
+                        output.extend(quote!(
+                            // Check if such namespace already exist, if so change its prefix to parent prefix
+                            let prefix_key = match serializer.parent_namespaces.get(#val) {
+                                Some(key) => key,
+                                None => #prefix_key,
+                            };
+                        ));
+                    }
+                    None => panic!("Prefix not defined: {}", prefix_key),
+                };
+
                 output.extend(quote!(
-                    field.attribute = Some(instant_xml::FieldAttribute::Prefix(#prefix_key));
+                    field.attribute = Some(instant_xml::FieldAttribute::Prefix(prefix_key));
+                ));
+            }
+            Some(FieldAttribute::Attribute) => {
+                is_attribute = true;
+                output.extend(quote!(
+                    serializer.add_attribute_key(&#name);
+                    field.attribute = Some(instant_xml::FieldAttribute::Attribute);
                 ));
-
-                if self.other_namespaces.get(&prefix_key).is_none() {
-                    missing_prefixes.insert(prefix_key);
-                };
             }
             _ => {}
         };
 
         output.extend(quote!(
-            self.#field_value.serialize(serializer, Some(&field))?;
+            serializer.set_field_context(field)?;
+            self.#field_value.serialize(serializer)?;
+            serializer.retrive_parent_default_namespace();
         ));
+
+        (output, is_attribute)
+    }
+
+    pub fn get_namespaces_token(&self) -> TokenStream {
+        let mut namespaces = quote!(
+            let mut to_remove: Vec<&str> = Vec::new();
+        );
+        for (k, v) in self.other_namespaces.iter() {
+            namespaces.extend(quote!(
+                // Only adding to HashMap if namespace do not exist, if it exist it will use the parent defined prefix
+                if let std::collections::hash_map::Entry::Vacant(v) = serializer.parent_namespaces.entry(#v) {
+                        v.insert(#k);
```

Comment:
You have a bunch of 8-column indents in this PR, please clean up the indentation (since `fmt` doesn't handle it inside the `quote!()` macro calls).

Reply:
done